plugins/repository-s3/src/main/java/org/elasticsearch/plugin/repository/s3/S3RepositoryPlugin.java

```diff
@@ -74,8 +74,16 @@ protected AwsS3Service createStorageService(Settings settings) {
     }
 
     @Override
+    public List<String> getSettingsFilter() {
+        return Arrays.asList(
+            S3Repository.Repository.KEY_SETTING.getKey(),
+            S3Repository.Repository.SECRET_SETTING.getKey());
```

Comment:
Yes. If I don't add those lines, it fails in the REST tests. See https://github.com/elastic/elasticsearch/blob/master/plugins/repository-s3/src/test/resources/rest-api-spec/test/repository_s3/20_repository.yaml